### PR TITLE
Validation Using New Observe API

### PR DESCRIPTION
**Pull request by rmorshea**
brief example

``` python

from traitlets import HasTraits, Int, observe, validate

class H(HasTraits):

    i = Int()
    j = Int()

    @validate('i')
    def _i_validator(self, change):
        return change['new']+1
    @observe('i')
    def _i_observer(self, change):
        self.j = change['new']

h  = H()
h.i = 1
print(h.i)
print(h.j)
```

print results:

```
2
2
```

### CODE REVIEW

**Comment by rmorshea** [comment]
depends on : ipython/traitlets#72

**Comment by SylvainCorlay** [comment]
It looks good to me. Maybe `unvalidate` is not the best name.

File: traitlets/traitlets.py [hunk 1]
```diff
@@ -650,8 +701,20 @@ def observe(*names):
     """
     return ObserveHandler(names)
 
+def validate(*names):
+    """ A decorator which can be used to cross validate members on a class.
```
Comment (SylvainCorlay) [inline traitlets/traitlets.py:705]: "... to cross validate trait attributes."?

**Comment by SylvainCorlay** [comment]
@rmorshea I just noticed you targeted this PR to my fork. 
I guess you can open it towards the main repo.

**Comment by rmorshea** [comment]
I'll make the pr against the main repo, and maybe "devalidate" instead of
"unvalidate"?

**Pull request closed without merging**